bin/pyspark

```diff
@@ -52,7 +52,7 @@ fi
 
 # Determine the Python executable to use for the executors:
 if [[ -z "$PYSPARK_PYTHON" ]]; then
-  if [[ $PYSPARK_DRIVER_PYTHON == *ipython* && $DEFAULT_PYTHON != "python2.7" ]]; then
+  if [[ $PYSPARK_DRIVER_PYTHON == *ipython* && $DEFAULT_PYTHON < "python2.7" ]]; then
```

Comment:
That said, the code in this PR is a strict improvement over the old overly-stringent check, so it's okay to merge it in its current form but I'd prefer to just make this super robust so we fix this once and for all.